### PR TITLE
DOC: update the pandas.Series.set_axis docstring

### PR DESCRIPTION
**Pull request by astrastefania**
Checklist for the pandas documentation sprint (ignore this if you are doing
an unrelated PR):

- [x] PR title is "DOC: update the <your-function-or-method> docstring"
- [x] The validation script passes: `scripts/validate_docstrings.py <your-function-or-method>`
- [x] The PEP8 style check passes: `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] The html version looks good: `python doc/make.py --single <your-function-or-method>`
- [x] It has been proofread on language by another sprint participant

Please include the output of the validation script below between the "```" ticks:

```


################################################################################
###################### Docstring (pandas.Series.set_axis) ######################
################################################################################

Assign desired index to given axis.

Indexes for columns or rows can be changed by assigning
a list-like or Index.

Parameters
----------
labels : list-like, Index
    The values for the new index.

axis : int, default 0
    The value 0 indentify the rows, 1 identify the columns.

inplace : boolean, default None
    Whether to return a new NDFrame instance.

    WARNING: inplace=None currently falls back to to True, but
    in a future version, will default to False.  Use inplace=True
    explicitly rather than relying on the default.

    .. versionadded:: 0.21.0
        The signature is make consistent to the rest of the API.
        Previously, the "axis" and "labels" arguments were respectively
        the first and second positional arguments.

Returns
-------
renamed : NDFrame or None
    An object of same type as caller if inplace=False, None otherwise.

See Also
--------
pandas.DataFrame.rename_axis : Alter the name of the index or columns.

Examples
--------
>>> s = pd.Series([1, 2, 3])
>>> s
0    1
1    2
2    3
dtype: int64
>>> s.set_axis(['a', 'b', 'c'], axis=0, inplace=False)
a    1
b    2
c    3
dtype: int64
>>> df = pd.DataFrame({"A": [1, 2, 3], "B": [4, 5, 6]})
>>> df.set_axis(['a', 'b', 'c'], axis=0, inplace=False)
   A  B
a  1  4
b  2  5
c  3  6
>>> df.set_axis(['I', 'II'], axis=1, inplace=False)
   I  II
0  1   4
1  2   5
2  3   6
>>> df.set_axis(['i', 'ii'], axis=1, inplace=True)
>>> df
   i  ii
0  1   4
1  2   5
2  3   6

################################################################################
################################## Validation ##################################
################################################################################

Errors found:
	Private classes (['NDFrame']) should not be mentioned in public docstring.
```

If the validation script still gives errors, but you think there is a good reason
to deviate in this case (and there are certainly such cases), please state this
explicitly.

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -487,24 +487,30 @@ def _expand_axes(self, key):
...
 
         Parameters
         ----------
-        labels: list-like or Index
-            The values for the new index
-        axis : int or string, default 0
+        labels : list-like, Index
+            The values for the new index.
+
+        axis : int, default 0
```
Comment (TomAugspurger) [inline pandas/core/generic.py]: axis : {0 or 'index', 1 or 'columns'}, default 0

Comment (TomAugspurger) [reply pandas/core/generic.py]: And modify the description to say that the string forms are also acepted.

Comment (astrastefania) [reply pandas/core/generic.py]: Thanks, so axis : {0 or 'index', 1 or 'columns'}, default 0 is enough or should I mentioned `int, str` too?

**Comment by astrastefania** [comment]
OK, I added the fixes mentioned :)

**Comment by TomAugspurger** [comment]
Pushed an update.

We weren't actually using the `set_axis` in a shared doc (I think). I converted it to a regular docstring. Going to let the tests finish before merging, in case I missed something. Running locally as well.

@astrastefania let me know if you see that the test finish before I do.

**Review by TomAugspurger — APPROVED** [review]

**Comment by astrastefania** [comment]
Hey @TomAugspurger, should I update anything in here? 
I'm happy to understand better how CircleCI works. :)

**Comment by TomAugspurger** [comment]
I think those failures are unrelated. Will look more closely tomorrow.

________________________________

**Comment by TomAugspurger** [comment]
@astrastefania those failures were fixed by https://github.com/pandas-dev/pandas/pull/20288. It failed on CircleCI since it tests your exact branch, which doesn't include those changes. It passed on TravisCI since it tests a merge commit, `master + your changes`, which does include those change (since they're on master).

We could merge master to this branch and repush and things should pass. But I'm just going to merge. Thanks!

**Pull request merged**